src/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.cpp

```diff
@@ -63,14 +73,36 @@ bool Propagator::isDefined() const
     return numericalSolver_.isDefined();
 }
 
+Size Propagator::getNumberOfCoordinates() const
+{
+    return this->coordinatesBrokerSPtr_->getNumberOfCoordinates();
+}
+
+const Shared<CoordinatesBroker>& Propagator::getCoordinatesBroker() const
```

Comment:
```suggestion
const Shared<CoordinatesBroker> Propagator::accessCoordinatesBroker() const
```

semantics here:
I think this should be named `accessCoordinatesBroker` since we're returning the object field directly and not a copy of it.